src/odemis/driver/simcam.py

```diff
@@ -258,7 +287,19 @@ def _generate(self):
             logging.debug("Sleeping extra %g s, for simulating event", extra_time)
             time.sleep(extra_time)
 
-        gen_img = self._simulate()
+        if self._img_simulator and self._mirror:
+            sim_img = self._img_simulator.simulate(self._mirror.position.value)
```

Comment:
Instead of calling this function here (instead of `_simulate()`), I think it should be within `_simulate()`, to replace `self._img`. In the case of `_mirror`, you can skip the `stage_shift` computation, but everything else of `_simulate()` can be done as is. This way, changing the binning, resolution, or exposure time will work.